pkg/policies/branch/branch.go

```diff
@@ -485,6 +588,12 @@ func mergeConfig(oc *OrgConfig, rc *RepoConfig, repo string) *mergedConfig {
 		if rc.BlockForce != nil {
 			mc.BlockForce = *rc.BlockForce
 		}
+		if rc.RequireUpToDateBranch != nil {
+			mc.RequireUpToDateBranch = *rc.RequireUpToDateBranch
+		}
+		if rc.RequireStatusChecks != nil {
```

Comment:
```
if len(rc.RequireStatusChecks) > 0 {
```
For clarity because it is a slice and not a pointer.

Reply:
Actually, the `!= nil` is intentional to add support for the case of stripping all the required checks defined at the org level at the repo level. The `yaml` package explicitly sets the attribute to `nil` instead of `[]string{}` when it's omitted.

E.g., one could have the config:
```yaml
requireStatusChecks: []
```

See https://go.dev/play/p/7RsvjHP1BY6

Should I add a comment here to make that clear?

Reply:
Ah, very cool! Yes, update the repoconfig comment on that to say that omitting leads to taking the orgconfig as is, but specifying an empty list will override the orgconfig to be empty. Thanks!

Reply:
Ok updated the comment: [c5b0001](https://github.com/ossf/allstar/pull/147/commits/c5b0001651646a1aeb95d710a90832266623e3f7)